cores/esp8266/Schedule.cpp

```diff
@@ -60,19 +75,45 @@ bool schedule_function(std::function<void(void)> fn)
         sLast->mNext = item;
     }
     sLast = item;
+
+    if (repeat_us)
+        item->callNow.reset(repeat_us);
+
     return true;
 }
 
+IRAM_ATTR // called from ISR
+bool schedule_function(std::function<void(void)> fn)
+{
+    return schedule_function_us([&fn](){ fn(); return false; }, 0);
+}
+
 void run_scheduled_functions()
```

Comment:
Good point.
We don't want to lock while executing the scheduled function themselvses.
One solution is to tag variables with `volatile`.
Another one is to build a local copy of the list while being locked, then unlock and run that list.

Reply:
I think I must step back on this.
Locking IRQ is the right thing to do when the value of a variable can be modified in a TAS block.
Even if there are registers holding some variables, they will not be changed while in a locked block because an IRQ won't occur in that block, regardless whether the variable is cached in a register.